sound/soc/sof/ipc4-topology.c

```diff
@@ -1387,6 +1466,80 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	return 0;
 }
 
+static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
+					   struct snd_pcm_hw_params *fe_params,
+					   struct snd_sof_platform_stream_params *platform_params,
+					   struct snd_pcm_hw_params *pipeline_params, int dir)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_soc_dapm_widget *widget = swidget->widget;
+	struct sof_ipc4_process *process = swidget->private;
+	struct sof_ipc4_available_audio_format *available_fmt = &process->available_fmt;
+	struct sof_ipc4_control_data *control_data;
+	struct snd_sof_control *scontrol = NULL;
+	void *cfg = process->ipc_config_data;
+	const struct snd_kcontrol_new *kc;
+	struct soc_bytes_ext *sbe;
+	void *data;
+	int ret, i;
+
+	available_fmt->ref_audio_fmt = &available_fmt->base_config->audio_fmt;
+
+	/*
+	 * Output format is optional for process modules.
+	 * Process modules setup the output format based on audio format tokens in topology.
+	 */
+	if (swidget->payload_with_output_fmt)
+		ret = sof_ipc4_init_audio_fmt(sdev, swidget, &process->base_config,
+					      &process->output_format, pipeline_params,
+					      available_fmt,
+					      sizeof(struct sof_ipc4_base_module_cfg));
+	else
+		ret = sof_ipc4_init_audio_fmt(sdev, swidget, &process->base_config,
+					      NULL, pipeline_params, available_fmt,
```

Comment:
nit-pick: the only difference between the two calls is ONE parameter, this could have been handled with a temporary variable and a single call.